lib/css/CssGenerator.js

```diff
@@ -5,37 +5,58 @@
+const { ReplaceSource, RawSource, ConcatSource } = require("webpack-sources");
+const { UsageState } = require("../ExportsInfo");
 const Generator = require("../Generator");
 const InitFragment = require("../InitFragment");
-const { CSS_TYPES } = require("../ModuleSourceTypesConstants");
+const { JS_TYPES, JS_AND_CSS_TYPES } = require("../ModuleSourceTypesConstants");
 const RuntimeGlobals = require("../RuntimeGlobals");
+const Template = require("../Template");
 
 /** @typedef {import("webpack-sources").Source} Source */
-/** @typedef {import("../../declarations/WebpackOptions").CssGeneratorExportsConvention} CssGeneratorExportsConvention */
-/** @typedef {import("../../declarations/WebpackOptions").CssGeneratorLocalIdentName} CssGeneratorLocalIdentName */
+/** @typedef {import("../../declarations/WebpackOptions").CssAutoGeneratorOptions} CssAutoGeneratorOptions */
+/** @typedef {import("../../declarations/WebpackOptions").CssGlobalGeneratorOptions} CssGlobalGeneratorOptions */
+/** @typedef {import("../../declarations/WebpackOptions").CssModuleGeneratorOptions} CssModuleGeneratorOptions */
 /** @typedef {import("../CodeGenerationResults")} CodeGenerationResults */
 /** @typedef {import("../Dependency")} Dependency */
 /** @typedef {import("../DependencyTemplate").CssDependencyTemplateContext} DependencyTemplateContext */
 /** @typedef {import("../DependencyTemplate").CssExportsData} CssExportsData */
 /** @typedef {import("../Generator").GenerateContext} GenerateContext */
 /** @typedef {import("../Generator").UpdateHashContext} UpdateHashContext */
+/** @typedef {import("../Module").ConcatenationBailoutReasonContext} ConcatenationBailoutReasonContext */
 /** @typedef {import("../Module").SourceTypes} SourceTypes */
 /** @typedef {import("../NormalModule")} NormalModule */
 /** @typedef {import("../util/Hash")} Hash */
 
 class CssGenerator extends Generator {
 	/**
-	 * @param {CssGeneratorExportsConvention} convention the convention of the exports name
-	 * @param {CssGeneratorLocalIdentName} localIdentName css export local ident name
-	 * @param {boolean} esModule whether to use ES modules syntax
+	 * @param {CssAutoGeneratorOptions | CssGlobalGeneratorOptions | CssModuleGeneratorOptions} options options
 	 */
-	constructor(convention, localIdentName, esModule) {
+	constructor(options) {
 		super();
-		this.convention = convention;
-		this.localIdentName = localIdentName;
-		/** @type {boolean} */
-		this.esModule = esModule;
+		this.convention = options.exportsConvention;
+		this.localIdentName = options.localIdentName;
+		this.exportsOnly = options.exportsOnly;
+		this.esModule = options.esModule;
+	}
+
+	/**
+	 * @param {NormalModule} module module for which the bailout reason should be determined
+	 * @param {ConcatenationBailoutReasonContext} context context
+	 * @returns {string | undefined} reason why this module can't be concatenated, undefined when it can be concatenated
+	 */
+	getConcatenationBailoutReason(module, context) {
```

Comment:
Before normal css modules cannot be concatenated, after union, normal css modules can be concatenated as well, after concatenated, original css modules are removed, and if so, will this affect chunk rendering as the css modules are already removed ?